### PR TITLE
feat: pin versions

### PR DESCRIPTION
**Pull request by tdejager**
This should also use latest rattler-build, and remove ruff duplicates.

Also removed solve-groups as they were not doing anything as of yet.

Ran tests and seemed to work.

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -11,22 +11,24 @@ build_sdist = "pixi run python -m build --sdist"
 
 [dependencies]
 python = ">=3.10"
-build = "*"
-rattler-build = "*"
-conda-build = "*"
-"ruamel.yaml" = "*"
+build = ">=0.7.0,<0.8"
+rattler-build = ">=0.18.1,<0.19"
```
Comment (baszalmstra) [inline pixi.toml:15]: This is for sure going to cause issues. We release minor bumps on a regular basis. Perhaps using `1.0` here makes sense ... ?

**Review by Hofer-Julian — APPROVED** [review]
Nice!

File: pixi.toml [hunk 2]
```diff
@@ -11,22 +11,21 @@ build_sdist = "pixi run python -m build --sdist"
 
 [dependencies]
 python = ">=3.10"
-build = "*"
-rattler-build = "*"
-conda-build = "*"
-"ruamel.yaml" = "*"
+build = ">=0.7.0,<0.8"
+rattler-build = ">=0.18.1,<1"
+conda-build = ">=24.3.0,<25.0"
```
Comment (nichmor) [inline pixi.toml:16]: wdyt about using this version selector?

`conda-build = "~=24.3"`

Comment (baszalmstra) [reply pixi.toml:16]: https://github.com/prefix-dev/pixi/issues/285

**Review by baszalmstra — APPROVED** [review]

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**